packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -46,21 +57,22 @@ export class PeerExchangeDiscovery
   private readonly peerExchange: WakuPeerExchange;
   private readonly options: Options;
   private isStarted: boolean;
-  private intervals: Map<PeerId, NodeJS.Timeout> = new Map();
+  private intervals: Map<string, NodeJS.Timeout> = new Map();
+  private queryAttempts: Map<string, number> = new Map();
```

Comment:
possible future improvement: extract the backoff and abort logic in a different class/function that tracks all of it. This class is starting to get cluttered.